### ISSUE TITLE
Why is xgboost great?

### ISSUE THREAD

**Issue by ThyrixYang** [dmlc/xgboost#5149]
Hi,

I want to show xgboost is strictly better than deep learning model such as MLP or CNN in some scenes. Could someone give some advice on dataset selection? The dataset would be better to be widely used in research papers.

Thank you.

**Comment by trivialfis** [comment]
There's no free lunch...

**Comment by ThyrixYang** [comment]
> There's no free lunch...

That means there must be some datasets that xgboost does better than deep model, I just want them.

**Comment by hcho3** [comment]
There are some domains where deep learning (neural networks) excels: computer visual, natural language processing, and reinforcement learning. These domains involves unstructured or semi-structured data (pixels, sequences, state spaces).

On the other hand, XGBoost is a good choice if you have **tabular** data, i.e. each feature has a well-defined meaning. Some reasons why you may want to choose XGBoost (or tree-based algorithms) over deep learning:

* No need to re-scale your data
* Less need for hyper-parameter tuning: in deep learning, you have to pick the right architecture, right hyper-parameters, and right learning rate schedule etc, etc.
* Less computation required for training
* Simpler model that's more interpretable. You can plot feature importances and even read tree outputs. Furthermore, SHAP (feature importance measure) is much faster to compute for tree models than deep learning models.

**Comment by trivialfis** [comment]
Actually tree models also excel at image segmentation and similar tasks, it's just currently XGBoost not being optimized for wide dataset.

**Comment by trivialfis** [comment]
See https://www.microsoft.com/en-us/research/publication/decision-forests-a-unified-framework-for-classification-regression-density-estimation-manifold-learning-and-semi-supervised-learning/ for some applications of tree models on image tasks.
